### PR TITLE
Info page user guide

### LINKED ISSUE #32: Add user guide at infopage
Add `user guide` at infopage.

### PR DESCRIPTION
**Pull request by csordasmarton**
Info page user guide with pluginable interface.

This commit resolves #32 issue.

### CODE REVIEW

File: webgui/userguide/userguide.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
+#
+# Useguide documentation for CodeCompass
+# Note: start every section id with 'userguide_' for uniqueness
```
Comment (bruntib) [inline webgui/userguide/userguide.md]: "userguide_" prefix doesn't ensure uniqueness.

File: webgui/userguide/userguide.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+The subtree of function callers can be opened recursively thus building a call
+chain of the given function.
+
+\subsubsection userguide_infotree_variables Variables
+Besides of the basic information about a variable (place of definition,
+qualified name, etc.) you can list the places of reading and writing of the
+given variable.
+
+\subsection userguide_infotree Info Tree in Source Code
```
Comment (bruntib) [inline webgui/userguide/userguide.md]: This section should go above the description of functions.

File: webgui/userguide/userguide.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+\subsubsection userguide_infotree_types Types
+Info tree contains type specific information if it is called on a class/struct
+name or enums.
+
+In case of a class, one can find information in the Info Tree about inheritance
+(from which classes this inherits and which classes inherit from this). The type
+of inheritance (public, protected, private) is also shown.
+
+Threw Info tree one can query the types which are aliases of the selected one,
```
Comment (bruntib) [inline webgui/userguide/userguide.md]: Threw -> Through

File: webgui/CMakeLists.txt [hunk 4]
```diff
@@ -13,4 +15,4 @@ install(FILES
 # Install npm packages
 install(CODE "set(CC_PACKAGE \"${CMAKE_CURRENT_SOURCE_DIR}/package.json\")")
 install(CODE "set(INSTALL_SCRIPTS_DIR \"${INSTALL_WEBROOT_DIR}/scripts/\")")
-install(SCRIPT InstallGUI.cmake)
+install(SCRIPT InstallGUI.cmake)
```
Comment (csordasmarton) [inline webgui/CMakeLists.txt]: This is a cmake script file which `concatenates the content of user guides` into a single file. This script also `calls the doxygen command` on the concatenated file. This file is being called when you run `make install`.

**Pull request merged**